spec/system/admin/enterprises_spec.rb

```diff
@@ -817,4 +817,170 @@
+        expect(page).not_to have_content "PROFILE ONLY"
+        expect(page).not_to have_content "HUB SHOP"
+
+        # Producer hub option is selected
+        page.find('a', class: 'selected', text: "PRODUCER HUB")
+        expect(enterprise.is_primary_producer).to eq true
+        expect(enterprise.reload.sells).to eq('any')
+
+        # Displays the correct dashboard sections
+        assert_hub_menu
+        assert_hub_content
+
+        # Changes to producer shop
+        page.find('a', text: "PRODUCER SHOP").click
+        click_button "Change now"
+        expect(page).to have_content update_message
+
+        # Checks changes are persistent
+        click_button "Change Package"
+
+        page.find('a', class: 'selected', text: "PRODUCER SHOP")
+        expect(enterprise.is_primary_producer).to eq true
+        expect(enterprise.reload.sells).to eq('own')
+
+        # Displays the correct dashboard sections
+        assert_hub_menu
+        assert_hub_content
+
+        # Changes to producer profile
+        page.find('a', text: "PRODUCER PROFILE").click
+        click_button "Change now"
+        expect(page).to have_content update_message
+
+        # Checks changes are persistent
+        click_button "Change Package"
+
+        page.find('a', class: 'selected', text: "PRODUCER PROFILE")
+        expect(enterprise.is_primary_producer).to eq true
+        expect(enterprise.reload.sells).to eq('none')
+
+        # Displays the correct dashboard sections
+        assert_supplier_menu
+        assert_supplier_content
+      end
+    end
+
+    context "via admin path, for a non-producer" do
+      before do
+        enterprise.update!(is_primary_producer: false)
+        visit spree.admin_dashboard_path
+      end
+
+      it "changes user role" do
+        click_button "Change Package"
+
+        # checks options for non-producer profile
+        expect(page).not_to have_content "PRODUCER PROFILE"
+        expect(page).not_to have_content "PRODUCER SHOP"
+        expect(page).not_to have_content "PRODUCER HUB"
+        expect(page).to have_content "PROFILE ONLY"
+        expect(page).to have_content "HUB SHOP"
+
+        # Producer hub option is selected
+        page.find('a', class: 'selected', text: "HUB SHOP")
+        expect(enterprise.reload.is_primary_producer).to eq false
+        expect(enterprise.reload.producer_profile_only).to eq false
+
+        # Displays the correct dashboard sections
+        assert_hub_menu
+        assert_hub_content
+
+        # Changes to producer shop
+        page.find('a', text: "PROFILE ONLY").click
+        click_button "Change now"
+        expect(page).to have_content update_message
+
+        # Checks changes are persistent
+        click_button "Change Package"
+
+        page.find('a', class: 'selected', text: "PROFILE ONLY")
+        expect(enterprise.reload.is_primary_producer).to eq false
+        expect(enterprise.reload.producer_profile_only).to eq false # this should be true?
```

Comment:
This test seems wrong ,  like the comment suggest, shouldn't it be : 
```suggestion
        expect(enterprise.reload.producer_profile_only).to eq true
```

Reply:
I agree - I would the `producer_profile_only` to change, when we select the `Profile only`. However, the app does not work in this way, I think this relates https://github.com/openfoodfoundation/openfoodnetwork/issues/7835.

So, if we proceed with the change you've proposed, we get:

```
Failures:

  1) 
    As an administrator
    I want to manage enterprises
 changing package via admin path, for a non-producer changes user role
     Failure/Error: expect(enterprise.reload.producer_profile_only).to eq true # this should be true?
     
       expected: true
            got: false
     
       (compared using ==)
     
       Diff:
       @@ -1 +1 @@
       -true
       +false
```

This attribute seems not to be decisive for the feature to work, all other assertions pass and reflect the correct behavior. However, this attribute _seems to be used_, as it is modified as expected the test on `line 921`, within the `context "via admin path, for a non-producer"`. I'm not sure we really can remove it, as proposed on #7835.

What do you think @rioug ?

Reply:
I don't know enough to decide if we should remove `producer_profile_only` or not. So I think at this stage it's probably better to keep the correct assertion, as you say it documents the behavior of the app. Maybe just add a comment to explain why we are checking if `producer_profile_only` is false.

Reply:
Ah, I've noticed now, that my initial comment was in the wrong place:
in line 921 we're in the context where the enterprise is not a producer, we should indeed get false; it is in the contexts:
  - `via admin path, for a producer`
  - `via enterprises path, for a producer`

And after changing to profile only, that `producer_profile_only` should be true, but it is never updated. It seems this is always false for enterprises (I've checked metabase and there are 4884 enterprises, all having `false` for `producer_profile_only`).

I've made the changes to reflect this on an additional PR, if you find it OK I'd then squash it.